collector/monitors/slashing_monitor.go

```diff
@@ -28,41 +29,74 @@ const (
 )
 
 type SlashingMonitor struct {
-	metrics              map[MetricName]float64
+	metrics              map[MetricName]MetricValue
 	metricVectors        map[MetricName]MetricVector
 	apiClient            *client.TerraLiteForTerra
 	validatorsRepository ValidatorsRepository
 	logger               *logrus.Logger
+	lock                 sync.RWMutex
+	flowManager          chan struct{}
 }
 
 func NewSlashingMonitor(cfg config.CollectorConfig, repository ValidatorsRepository) *SlashingMonitor {
 	m := &SlashingMonitor{
-		metrics:              make(map[MetricName]float64),
+		metrics:              make(map[MetricName]MetricValue),
 		metricVectors:        make(map[MetricName]MetricVector),
 		apiClient:            cfg.GetTerraClient(),
 		validatorsRepository: repository,
 		logger:               cfg.Logger,
+		lock:                 sync.RWMutex{},
 	}
 
+	if cfg.SlashingMonitorUpdateInterval > 0 {
```

Comment:
I think this is a bit overengineered. This is sort of the standard way to do such things:

```
func NewSlashingMonitor(cfg config.CollectorConfig, repository ValidatorsRepository) *SlashingMonitor {
	m := &SlashingMonitor{
		metrics:              make(map[MetricName]MetricValue),
		metricVectors:        make(map[MetricName]MetricVector),
		apiClient:            cfg.GetTerraClient(),
		validatorsRepository: repository,
		logger:               cfg.Logger,
		lock:                 sync.RWMutex{},
	}

	if cfg.SlashingMonitorUpdateInterval > 0 {
		go m.updateRoutine(ctx, time.Second*time.Duration(cfg.SlashingMonitorUpdateInterval))
	}

	return m
}

func (m *SlashingMonitor) updateRoutine(ctx context.Context, updateInterval time.Duration) {
	tk := time.NewTicker(updateInterval)
	for {
		select {
		case <-tk.C:
			// Move the mutex inside updateData.
			err := m.updateData(context.Background())
			if err != nil {
				m.logger.Errorln("failed to update SlashingMonitor data:", err)
			}
		case <-ctx.Done():
			return
		}
	}
}
```